### DISCUSSION TITLE
fail speedtest on UE

### DISCUSSION THREAD

**Discussion by initChat** [srsran/srsRAN_Project#630] [General Help]
I have problem when using Raspberry Pi with Quectel board as UE. When I try to do `speedtest-cli`, the open5gs screen shows that the gNB-UE is disconnected before reconnecting. Although I can retrieve speedtest-cli result, I lost connection with the srsRAN network and need to reconnect. Please advise me What should I do?

This open5gs log shows UE disconnect from the network and try to reconnect.
![Screenshot from 2024-05-24 16-47-25](https://github.com/srsran/srsRAN_Project/assets/40858099/e65a78d7-98de-414b-97fb-7e8f4d0ddcc2)

Previously, I successfully use CPE as UE connecting to the srsRAN network. The CPE can provide internet for a mobile phone. Then, I change to Raspberry Pi with Quectel board. The RPI with Quectel can receive IP address and be able to ping to `1.1.1.1, 8.8.8.8, www.chula.ac.th`. But cannot do `speedtest-cli` without lost internet connection.

**system**
Core network: open5gs
RAN: USRP B210
UE: Raspberry PI 3 with Quectel RM510Q-GL

**gnb config**
[gnb_config.txt](https://github.com/srsran/srsRAN_Project/files/15431806/gnb_config.txt)

**log**
- gnb_log [gnb.zip](https://github.com/srsran/srsRAN_Project/files/15431655/gnb.zip)
- [gnb_wireshark.zip](https://github.com/srsran/srsRAN_Project/files/15431659/gnb_wireshark.zip)
- open5gs_wireshark_result [ogstun_core_network.zip](https://github.com/srsran/srsRAN_Project/files/15431662/ogstun_core_network.zip)
- speedtest_result_on_rpi [rpi_speedtest.txt](https://github.com/srsran/srsRAN_Project/files/15431673/rpi_speedtest.txt)

**Comment by pgawlowicz** [comment]
In gnb logs, you have:
```
2024-05-24T09:26:59.697188 [DU-MNG  ] [I] ue=0: RLF timer expired with cause="RLC max ReTxs reached". Requesting a UE release...
2024-05-24T09:26:59.697202 [CU-CP-F1] [I] Rx PDU GNB-DU-ID=1 GNB-DU-UE-F1AP-ID=0: UEContextReleaseRequest
```

could you try to tune TX/RX gains, both seem to be quite high. Maybe the signal gets saturated.

**Reply by pgawlowicz** [reply]
@pchat-imm any update on this issue?

**Reply by initChat** [reply]
currently try `tx_gain: 45, rx_gain: 30` but not success in testing `speedtest`. Will be able to test again on tomorrow or next two weeks. Will try to update as soon as possible. Thank you

**Reply by pgawlowicz** [reply]
@pchat-imm any update on this issue?
